core/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -429,10 +434,21 @@ public FetchSearchResult executeFetchPhase(ShardFetchRequest request, SearchTask
             throw ExceptionsHelper.convertToRuntime(e);
         } finally {
             cleanContext(context);
+            //Release searchContext if we have no more hits
+            if((context.queryResult().from() + context.queryResult().size()) >= 5){
```

Comment:
`context.scrollContext().totalHits` gives us only the total hits in that shard and not the index. Or did i use the wrong api?